packages/react/src/components/StructuredList/StructuredList.js

```diff
@@ -127,32 +126,38 @@ StructuredListBody.defaultProps = {
   onKeyDown: () => {},
 };
 
+function useFocusWithin() {
+  const [hasFocusWithin, setHasFocusWithin] = useState(false);
+  return [hasFocusWithin, setHasFocusWithin];
+}
+
 export function StructuredListRow(props) {
-  const {
-    onKeyDown,
-    tabIndex,
-    children,
-    className,
-    head,
-    label,
-    ...other
-  } = props;
+  const { onKeyDown, children, className, head, ...other } = props;
+  const [hasFocusWithin, setHasFocusWithin] = useFocusWithin();
```

Comment:
```suggestion
const [hasFocusWithin, setHasFocusWithin] = useState(false);
```

Based on note above, if we want to drop the custom hook we would just update this to

Reply:
I think the custom hook is helpful! I'll keep it and moved it to /internal

Reply:
I think if it's a wrapper around `useState` then we might want to just embed it directly into the component (it doesn't add much value as just a wrapper).

An example of a custom hook may be something like this: https://codesandbox.io/s/focuswithin-wtzp1 that could be generalized